app/views/charge.njk

```diff
@@ -349,7 +349,13 @@
                     {{ __p("commonConjunctions.or") }}
                   </span>
                   <img src="/images/amex-security-code.png" class="amex-cvc hidden" alt="{{ __p("cardDetails.amexcvcTip") }}"/>
-                </div>
+                 </div>
+                   {% if savePaymentInstrumentToAgreement and agreementId  and agreementDescription %}
+                   <div>
```

Comment:
I think the spec calls for this snippet to always be shown by the continue button, closest to the users next action. Here will insert it above collecting billing address (if required by the account) so I think it should move down